plugins/meta/firewall/nftables.go

```diff
@@ -0,0 +1,350 @@
+				"iifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"oifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		} else {
+			rule = []string{
+				"insert", "rule", nb.targetTable, nb.targetChain,
+				"position", fmt.Sprintf("%d", nb.targetHandle),
+				"iifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"oifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		}
+		nb.newRules = append(nb.newRules, rule)
+	}
+
+	if !isInboundInterfaceRuleExists {
+		requireChange = true
+		if len(nb.rules) == 0 {
+			rule = []string{
+				"add", "rule", nb.targetTable, nb.targetChain,
+				"oifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"ip", "daddr", nb.targetNetwork.String(),
+				"ct", "state", "established,related",
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		} else {
+			rule = []string{
+				"insert", "rule", nb.targetTable, nb.targetChain,
+				"position", fmt.Sprintf("%d", nb.targetHandle),
+				"oifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"ip", "daddr", nb.targetNetwork.String(),
+				"ct", "state", "established,related",
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		}
+		nb.newRules = append(nb.newRules, rule)
+	}
+
+	if !isOutboundInterfaceRuleExists {
+		requireChange = true
+		if len(nb.rules) == 0 {
+			rule = []string{
+				"add", "rule", nb.targetTable, nb.targetChain,
+				"iifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"ip", "saddr", nb.targetNetwork.String(),
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		} else {
+			rule = []string{
+				"insert", "rule", nb.targetTable, nb.targetChain,
+				"position", fmt.Sprintf("%d", nb.targetHandle),
+				"iifname", fmt.Sprintf("\"%s\"", nb.targetInterface),
+				"ip", "saddr", nb.targetNetwork.String(),
+				"counter", "packets", "0", "bytes", "0", "accept",
+			}
+		}
+		nb.newRules = append(nb.newRules, rule)
+	}
+
+	if requireChange {
+		return true
+	}
+	return false
+}
+
+func (nb *nftBackend) addRules() error {
+	if len(nb.newRules) == 0 {
+		return nil
+	}
+	for _, cmdArgs := range nb.newRules {
+		stdoutLines, stderrLines, err := nb.execCommand(cmdArgs)
+		if err != nil {
+			return fmt.Errorf(
+				"encountered error %s: %s (stdout: %s, stderr: %s)",
+				cmdArgs, err, stdoutLines, stderrLines,
+			)
+		}
+	}
+	return nil
+}
+
+func (nb *nftBackend) delRules() error {
+	if len(nb.markedDelRules) == 0 {
+		return nil
+	}
+	return nil
+}
+
+func (nb *nftBackend) isValidInput(result *current.Result) error {
+	if len(result.Interfaces) == 0 {
+		return fmt.Errorf("the data passed to firewall plugin did not contain network interfaces")
+	}
+
+	if result.Interfaces[0].Name == "" {
+		return fmt.Errorf("the data passed to firewall plugin has no bridge name, e.g. cnibr0")
+	}
+
+	nb.targetInterface = result.Interfaces[0].Name
```

Comment:
I don't think target interface is always at position 0, what we want here is the peer of container interface.